plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/FacetUtil.java

```diff
@@ -68,7 +68,7 @@
   Set<IFacetedProject.Action> facetInstallSet = new HashSet<>();
 
   public FacetUtil(IFacetedProject facetedProject) {
-    Preconditions.checkNotNull(facetedProject);
+    Preconditions.checkNotNull(facetedProject, "project not faceted");
```

Comment:
This isn't quite true: it was passed `null`.  It may be that the caller tried to get the faceted project which didn't exist.